### ISSUE TITLE
Expose Body Water Percentage

### ISSUE THREAD

**Issue by Goalpwr** [home-assistant/home-assistant.github.io#30684]
### Feedback

TLDR: Please expose body water percentage.

Quick ask – could we get the body water percentage in the Withings Muscle and Body Fat calculation? Right now, the muscle and fat numbers are cool, but they swing a lot depending on body water. Case in point: at 55% body water, my fat mass is 13.9. Bump that water percentage to 55.5%, and fat mass drops to 13.2. Hit 57.1% water, and bam, fat mass is down to 11.1. 

This happened over just a few days with no big weight changes. Adding body water percentage would help us all be more aware of the accuracy of the trend especially if we're able to keep our body water consistent (or at least know what it is in a home assitant dashboard).  Thanks!

### URL

https://www.home-assistant.io/integrations/withings/

### Version

2024.1.2

**Comment by joostlek** [comment]
Hey!

The Withings API does not return the water mass. It does return cellular water in kg. But that is only for specific scales iirc.

All measurement types are checkable in this page: https://developer.withings.com/api-reference/#tag/measure/operation/measure-getmeas

This means we can't return the body water percentage.

**Comment by Goalpwr** [comment]
> Hey!
> 
> The Withings API does not return the water mass. It does return cellular water in kg. But that is only for specific scales iirc.
> 
> All measurement types are checkable in this page: https://developer.withings.com/api-reference/#tag/measure/operation/measure-getmeas
> 
> This means we can't return the body water percentage.

Could you expose the cellular water in kg,  I'll I should be able to work with that, please let me know thanks!

**Comment by joostlek** [comment]
We already do. If Withings exposes it, we add it

**Comment by Goalpwr** [comment]
> We already do. If Withings exposes it, we add it

Hmm interesting, either I'm doing something wrong in home assistant, or my Body Cardio scale, which measures water mass, isn't providing the data to the integration.  I see the field but in home assistant, but it's not used
![withingsimage](https://github.com/home-assistant/home-assistant.io/assets/21008881/bb2457bf-dfc3-494b-aa5e-9badc79eacc1).  If it's me please let me know oh and thank you for providing this extremely useful integration.

**Comment by joostlek** [comment]
I think the water percentage is different than the cellular water

**Comment by Goalpwr** [comment]
@joostlek Occam's razor............  I just needed to click enable.  That resolved the issue.  Thank You for your responses and patience.

**Comment by joostlek** [comment]
No problem, enjoy!
